src/pipelines/sentence_embeddings/pipeline.rs

```diff
@@ -0,0 +1,370 @@
+        } = self.tokenize(inputs);
+        let tokens_ids = Tensor::stack(&tokens_ids, 0).to(self.var_store.device());
+        let tokens_masks = Tensor::stack(&tokens_masks, 0).to(self.var_store.device());
+
+        let (tokens_embeddings, all_attentions) =
+            self.transformer.forward(&tokens_ids, &tokens_masks)?;
+
+        let mean_pool = self.pooling_layer.forward(tokens_embeddings, &tokens_masks);
+        let maybe_linear = if let Some(dense_layer) = &self.dense_layer {
+            dense_layer.forward(&mean_pool)
+        } else {
+            mean_pool
+        };
+        let maybe_normalized = if self.normalize_embeddings {
+            let norm = &maybe_linear
+                .norm_scalaropt_dim(2, &[1], true)
+                .clamp_min(1e-12)
+                .expand_as(&maybe_linear);
+            maybe_linear / norm
+        } else {
+            maybe_linear
+        };
+
+        Ok(SentenceEmbeddingsModelOuput {
+            embeddings: maybe_normalized,
+            all_attentions,
+        })
+    }
+
+    pub fn encode<S>(&self, inputs: &[S]) -> Result<Vec<Embedding>, RustBertError>
+    where
+        S: AsRef<str> + Sync,
+    {
+        let SentenceEmbeddingsModelOuput { embeddings, .. } = self.forward(inputs)?;
+        Ok(Vec::from(embeddings))
+    }
+
+    fn nb_layers(&self) -> usize {
+        use SentenceEmbeddingsOption::*;
+        match (&self.transformer, &self.transformer_config) {
+            (Bert(_), ConfigOption::Bert(conf)) => conf.num_hidden_layers as usize,
+            (Bert(_), _) => unreachable!(),
+            (DistilBert(_), ConfigOption::DistilBert(conf)) => conf.n_layers as usize,
+            (DistilBert(_), _) => unreachable!(),
+            (Roberta(_), ConfigOption::Bert(conf)) => conf.num_hidden_layers as usize,
+            (Roberta(_), _) => unreachable!(),
+            (Albert(_), ConfigOption::Albert(conf)) => conf.num_hidden_layers as usize,
+            (Albert(_), _) => unreachable!(),
+            (T5(_), ConfigOption::T5(conf)) => conf.num_layers as usize,
+            (T5(_), _) => unreachable!(),
+        }
+    }
+
+    fn nb_heads(&self) -> usize {
+        use SentenceEmbeddingsOption::*;
+        match (&self.transformer, &self.transformer_config) {
+            (Bert(_), ConfigOption::Bert(conf)) => conf.num_attention_heads as usize,
+            (Bert(_), _) => unreachable!(),
+            (DistilBert(_), ConfigOption::DistilBert(conf)) => conf.n_heads as usize,
+            (DistilBert(_), _) => unreachable!(),
+            (Roberta(_), ConfigOption::Roberta(conf)) => conf.num_attention_heads as usize,
+            (Roberta(_), _) => unreachable!(),
+            (Albert(_), ConfigOption::Albert(conf)) => conf.num_attention_heads as usize,
+            (Albert(_), _) => unreachable!(),
+            (T5(_), ConfigOption::T5(conf)) => conf.num_heads as usize,
+            (T5(_), _) => unreachable!(),
+        }
+    }
+
+    pub fn encode_with_attention<S>(
```

Comment:
Is there a use case to justify the added complexity? As far as I know the reference `sentence-transformers` only has an `encode` method: https://www.sbert.net/docs/package_reference/SentenceTransformer.html

Reply:
Right, similarly to my other comment, this has been helpful for explainability/vizualisation features in our use-cases.
I think that being able to get attentions one way or another is important with this kind of models.